### DISCUSSION TITLE
what changes would i have to make if i have to point it towards multiple buckets ?

### DISCUSSION THREAD

**Discussion by akashgreninja** [nginxinc/nginx-s3-gateway#221] [Q&A]
at the moment i want to cache  contents of 3+ buckets or access them atleast how do i change th4 standalone ubuntu such that i can point it towards 3+ buckets ?

**Comment by 4141done** [comment]
The gateway does not currently support multiple buckets. The easiest way to accomplish this is to use docker.  You can see a discussion of this here https://github.com/nginxinc/nginx-s3-gateway/discussions/153

If you cannot use docker it becomes more complicated and you'll have to make changes to https://github.com/nginxinc/nginx-s3-gateway/blob/master/common/etc/nginx/nginx.conf
and
https://github.com/nginxinc/nginx-s3-gateway/blob/master/common/etc/nginx/templates/default.conf.template

To reference the same js files for multiple servers.  I'm not able to provide full instructions here but if anyone else has done it feel free to answer.

**Comment by akashgreninja** [comment]
So this might not exactly be an ideal answer here is what i did
i shifted from environment to js_val
and added 2 server blocks with the same logic but different js_val
so basically now
https://localhost:80/ is one bucket
https://localhost:8080/ is another

i used the standalone script btw
this can be changed into one port by adding different server_name var
